### PR TITLE
Master project planner jas

### PR DESCRIPTION
**Pull request by jem-odoo**
Planner for project module : help people to start with the module, and setup correctly !
- Task : https://www.odoo.com/web#id=8874&view_type=form&model=project.task&action=333&active_id=131
- Pad : https://pad.odoo.com/p/r.IdqW6PopC1KXCHQX

### CODE REVIEW

File: addons/planner/static/src/less/planner.less [hunk 1]
```diff
@@ -204,6 +204,172 @@
         margin-left: auto;
         margin-right: auto;
 
+        .trophy{
```
Comment (jem-odoo) [inline addons/planner/static/src/less/planner.less]: I'm ok with this deplacement, but try to reduce the less. Remove some attributes, even if it change a little the complet design. 700 lines of less, AL will not be happy ^^

File: addons/planner_project/data/planner_data.xml [hunk 2]
```diff
@@ -0,0 +1,1211 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<odoo>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: :+1:

File: addons/planner_project/data/planner_data.xml [hunk 3]
```diff
@@ -0,0 +1,1211 @@
...
+                            <span class="fa fa-fw"/>Generate Reports
+                        </a>
+                    </li>
+                </ul>
+            </div>
+
+            <!--Planner Pages-->
+            <div class="pages col-sm-8 col-md-9 col-xs-12">
+
+                <div id="planner_page1" class="planner-page visible">
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: rebase your branch, I change classes : now using bootstrap `show` class instead of visible ;)

File: addons/planner_project/data/planner_data.xml [hunk 4]
```diff
@@ -0,0 +1,1211 @@
...
+                                            <li> Keep track of messages and conversations</li>
+                                            <li> List, plan and track things to do</li>
+                                            <li> Share files and manage versions</li>
+                                            <li> View statistics (time spent, efficiency, etc.)</li>
+                                            <li> Know what my employees are working on</li>
+                                        </ul>
+                                    </div>
+                                </li>
+
+                                <li><span>#1</span><textarea id="input_element_exp1" value="" line="1" type="text"/></li>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: Are the `span` tag required ?

File: addons/planner_project/data/planner_data.xml [hunk 5]
```diff
@@ -0,0 +1,1211 @@
...
+                                        <h5>Examples</h5>
+                                        <ul>
+                                            <li> Customer support tickets</li>
+                                            <li> Software development</li>
+                                            <li> Consultancy mission</li>
+                                            <li> Communication campaign</li>
+                                        </ul>
+                                    </div>
+                                </li>
+                                <li><span>#1</span><textarea id="input_element_kpi1" value="" line="1" type="text"/></li>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: Same as above

File: addons/planner_project/data/planner_data.xml [hunk 6]
```diff
@@ -0,0 +1,1211 @@
...
+                            </div>
+                        </div>
+                        <div class="row">
+                            <div class="col-sm-1 col-md-2 icon"><span data-icon="&#xe04a;" /></div>
+                            <div class="col-xs-12 col-md-10">
+                                <p>
+                                    <strong>Rely on the Chatter</strong><br/>
+                                    Below every Task and Issue (or more generally, below every Document in Odoo) is an area called Chatter.
+                                </p>
+                                <p>It's for logging every change, event or message related to the Docu
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: "Selling" for project ?

File: addons/planner_project/data/planner_data.xml (hunk 3 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: Done!

File: addons/planner_project/data/planner_data.xml (hunk 4 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: Yes! without <span>, design and font changes.

File: addons/planner_project/data/planner_data.xml (hunk 6 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: Oh! Sorry
Done.

File: addons/planner/static/src/less/planner.less (hunk 1 above)
Comment (jas-odoo) [inline addons/planner/static/src/less/planner.less]: Done!

File: addons/planner/static/src/js/planner.js [hunk 7]
```diff
@@ -64,8 +64,9 @@
         setup: function(planner){
             var self = this;
             this.planner = planner;
+            this.dialog && this.dialog.destroy();
             this.dialog = new instance.planner.PlannerDialog(this, planner);
-            this.$(".oe_planner_progress").tooltip({html: true, title: this.planner.tooltip_planner, placement: 'bottom', delay: {'show': 500}});
+            this.$(".oe_planner_progress").tooltip({html: true, placement: 'bottom', delay: {'show': 500}}).attr('data-original-title', this.planner.tooltip_planner);
```
Comment (jem-odoo) [inline addons/planner/static/src/js/planner.js]: Why this change ? The 'title' key of the dict wasn't working ? can you explain ? Thanks

File: addons/planner/static/src/js/planner.js [hunk 8]
```diff
@@ -267,7 +269,7 @@
             var btn = $(ev.currentTarget);
             var page_id = btn.attr('data-pageid');
             var active_menu = self.$(".oe_planner li a[href=#"+page_id+"] span");
-            var active_page = self.$(".oe_planner div[id^='planner_page'].panel-collapse.collapse.in");
+            var active_page = self.$(".oe_planner div[id^='planner_page'].planner-page.show");
```
Comment (jem-odoo) [inline addons/planner/static/src/js/planner.js]: Is `show` a native bootstrap class ?

File: addons/planner_project/data/planner_data.xml [hunk 9]
```diff
@@ -0,0 +1,1211 @@
...
+                                        <h5>Examples</h5>
+                                        <ul>
+                                            <li> Customer support tickets</li>
+                                            <li> Software development</li>
+                                            <li> Consultancy mission</li>
+                                            <li> Communication campaign</li>
+                                        </ul>
+                                    </div>
+                                </li>
+                                <li><span>#1</span><textarea id="input_element_act1" value="" line="1" type="text"/></li>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: Is this required to put the number in a `span` tag ?

File: addons/planner_project/data/planner_data.xml [hunk 10]
```diff
@@ -0,0 +1,1211 @@
...
+                                </div>
+                            </div>
+                        </div>
+                        <div class="row">
+                            <div class="col-md-12">
+                                <div class="panel panel-success">
+                                    <div class="panel-heading">
+                                        <span class="panel-title">
+                                            <span class="fa fa-exclamation-circle"/>
+                                            <strong> Issues</strong>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: Is the `strong` tag required since the text has the class `panel-title` ?

File: addons/planner/static/src/js/planner.js (hunk 7 above)
Comment (jas-odoo) [inline addons/planner/static/src/js/planner.js]: If more than one planner are installed at same time, then one planner tool tip is clash with other planner tool tip. By setting 'original-title', tool tips doesn't clash.

File: addons/planner_project/models/planner_project.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    _inherit = 'planner.planner'
+
+    @api.model
+    def _get_planner_application(self):
+        planner = super(PlannerProject, self)._get_planner_application()
+        planner.append(['planner_project', 'Project Planner'])
+        return planner
+
+    @api.model
+    def _prepare_planner_project_data(self):
```
Comment (jem-odoo) [inline addons/planner_project/models/planner_project.py]: This method is not required, since the render method already put `prepare_backend_url`in its default value. If project planner should have a 'prepare' method, this will only add specific value to project planner only.

File: addons/planner_project/data/planner_data.xml (hunk 9 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: Yes. css is written in planner.js for its position and its various attribute.

File: addons/planner_project/data/planner_data.xml [hunk 12]
```diff
@@ -0,0 +1,1211 @@
...
+                                    <input class="col-xs-12 col-sm-7 col-sm-offset-1" id="input_element_stage_10" type="text" value="Reason for cancellation has been documented"/>
+                                </li>
+                                <li class="row">
+                                    <span>#5</span>
+                                    <input class="col-xs-12 col-sm-4 user_project" data-id="input_element_user_stage4" id="input_element_stage_4" type="text"/>
+                                    <input class="col-xs-12 col-sm-7 col-sm-offset-1" id="input_element_stage_11" type="text"/>
+                                </li>
+                                <li class="row">
+                                    <span>#6</span>
+                                    <input class="col-xs-12 col-sm-4 user_project" data-id="input_element_user_stage5" id="input_element_stage_5" type="text"/>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: why do you need `data-id` and `id` ?

File: addons/planner_project/static/src/js/planner_project.js [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+    "use strict";
+    var _t = openerp.web._t;
+
+    openerp.planner.PlannerDialog.include({
+        prepare_planner_event: function() {
+            var self = this;
+            this._super.apply(this, arguments);
+            if(self.planner['planner_application'] == 'planner_project') {
+                var stages = {
+                    'development_process': [
```
Comment (jem-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: for each method (development_process, marketing_department, ....) should it be better to have a dict matching the id of the input with the value ? 
Such as `{'input_element_stage_1' : _t('validation'), 'input_element_stage_2' : _t('Development'), .... }`.
I think this change can also be apply to planner_crm ;)

File: addons/planner_project/static/src/less/planner_project.less [hunk 14]
```diff
@@ -0,0 +1,62 @@
+
+#PlannerModal {
```
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: I guess this style is specific for planner project. To avoid conflict with other planner (since the page has the same name), add a class `planner_xxxxx` (where xxxx is the planner name, such as in the selection field of the planner model) on the tag `#PlannerModal`.

Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Since the #PlannerModal id is in a js qweb template, put the new `planner_xxxxx` class with the `oe_planner` class in the view directly to not add it with javascript ;)

**Comment by jem-odoo** [comment]
Keep 2 commits for this merge : 
* [IMP] planner, planner_crm : modifications to prepare other planner (the less style move, some change in the js, ....)
* [ADD] planner_project : ....
Thanks :smile_cat:

File: addons/planner_project/data/planner_data.xml [hunk 15]
```diff
@@ -0,0 +1,1211 @@
...
+                                    <p><strong>4. Use Status Indicator</strong><br/>
+                                        The Status indicator helps you manage Tasks and Issues by attributing them one of 3 different colour: Grey, Green or Red. The meaning of these Status can be freely configured, for example:
+                                        <li>Grey: the Task is in progress (someone is working on it)</li>
+                                        <li>Green: the Task is ready for next stage (the job for this stage is complete)</li>
+                                        <li>Red: the Task is blocked (there's a problem)</li></p>
+                                    <p>To configure these Kanban Status, got to the 'Project Stages' and click on a Stage.</p>
+                                </div>
+                            </div>
+                            <div class="row">
+                                <div class="col-sm-1 col-md-2 icon"><span data-icon="&#xe017;" /></div>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: icon not well rendered on chrome Version 40.0.2214.115 (64-bit). I see a square instead :crying_cat_face:

File: addons/planner_project/data/planner_data.xml [hunk 16]
```diff
@@ -0,0 +1,1211 @@
...
+                        <li><strong>In 'Pull' mode</strong>, tasks ready to progress to the next stage are just marked  as 'Ready for next stage (using the status icon) by the person responsible for the current stage. Then, the person responsible for the next stage  takes the task and moves it to the next stage. This is the best way to work if you have diluted responsibilities for your Kanban stages.</li>
+                    </div>
+                </div>
+
+                <div id="planner_page11" class="planner-page">
+                    <h1>Notifications</h1>
+                    <p class="text-muted">Managing notifications is essential: too few and you risk missing critical information, too many and you will be  overloaded with unnecessary information. The trick is to find the right balance between the projects, stages and tasks you want to be informed about. Fortunately, Odoo Project has many levels of notifications and messages you can choose from.</p>
+                    <div class="container-fluid paragraphs ">
+                        <div class="row">
+                            <div class="col-sm-1 col-md-2 icon"><span data-icon="&#xe051;" /></div>
```
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: Same here, icon not rendered :s

File: addons/planner_project/static/src/less/planner_project.less [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+    font-family: "Lato", serif;
+
+    #planner_page1 {
+        h1 {margin-bottom: 20px;}
```
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: remove this. It is useless.

File: addons/planner_project/static/src/less/planner_project.less [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+                &:first-child {
+                    padding-left: 0;
+                }
+                &:last-child {
+                    padding-right: 0;
+                }
+            }
+        }
+    }
+    #planner_page4, #planner_page10 {
```
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: This section is useless too : the font size stay the same without the style on `li`, and there is no `select` in those pages.

File: addons/planner_project/data/planner_data.xml (hunk 10 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: As 'panel-title' class doesn't have bold attribute. So i have use strong tag. Should i remove strong tag?

File: addons/planner_project/static/src/less/planner_project.less [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+            margin-bottom: 8px;
+        }
+    }
+    #planner_page2,#planner_page3 {
+        textarea{
+            width: 100%;
+            max-width: 100%;
+        }
+    }
+    #planner_page4, #planner_page7, #planner_page8, #planner_page11, #planner_page14 {
```
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: not applicable for #planner_page4

File: addons/planner_project/static/src/less/planner_project.less [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    #planner_page4, #planner_page10 {
+        li {
+            font-size: 14px;
+        }
+        select {
+            font-size: 14px;
+        }
+    }
+    #planner_page11 {
```
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: not `inputbox` on that page, and we can survive without the `padding: 5px` so you can remove it

File: addons/planner_project/static/src/less/planner_project.less (hunk 17 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Done

File: addons/planner_project/static/src/less/planner_project.less (hunk 19 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Yup.Sorry
Done

File: addons/planner/static/src/js/planner.js (hunk 8 above)
Comment (jas-odoo) [inline addons/planner/static/src/js/planner.js]: yes.

File: addons/planner_project/models/planner_project.py (hunk 11 above)
Comment (jas-odoo) [inline addons/planner_project/models/planner_project.py]: ok.Done

File: addons/planner_project/static/src/less/planner_project.less (hunk 20 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: done

File: addons/planner_project/static/src/less/planner_project.less (hunk 18 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Done..but there is change in fonts.is it ok?

File: addons/planner_project/static/src/less/planner_project.less (hunk 14 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Done!
Can the same thing be done in crm too?

File: addons/planner_project/data/planner_data.xml (hunk 12 above)
Comment (jas-odoo) [inline addons/planner_project/data/planner_data.xml]: data-id is used to put current value to responsibility 'kanban stage'.Anyway i have make code working by removing data-id attr.

File: addons/planner/static/src/js/planner.js (hunk 7 above)
Comment (jem-odoo) [inline addons/planner/static/src/js/planner.js]: I think `this.$(".oe_planner_progress").tooltip('destroy').tooltip({html: true, title: this.planner.tooltip_planner, placement: 'bottom', delay: {'show': 500}});` can do the job  and it's cleaner ;)

File: addons/planner_project/data/planner_data.xml (hunk 10 above)
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: I thought there was a bold attribute on the class. You can keep it !

File: addons/planner_project/data/planner_data.xml (hunk 15 above)
Comment (jem-odoo) [inline addons/planner_project/data/planner_data.xml]: My bad, it was a cache mistake on my machine ^^

File: addons/planner_project/static/src/less/planner_project.less (hunk 14 above)
Comment (jem-odoo) [inline addons/planner_project/static/src/less/planner_project.less]: Yep, do the same for crm_planner ;) Thanks !

File: addons/planner/static/src/js/planner.js (hunk 7 above)
Comment (jas-odoo) [inline addons/planner/static/src/js/planner.js]: yup.done

File: addons/planner_project/static/src/js/planner_project.js (hunk 13 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: @jem-odoo : Current code for this works fine. The given solution will enlarge the code. Should I keep it as it or make according to {'input_element_stage_1' : _t('validation'), 'input_element_stage_2' : _t('Development'), .... }

File: addons/planner_project/static/src/js/planner_project.js [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+                    }
+                }
+                self.$el.on('change', '#input_element_kanban_stage_pipeline', function(ev) {
+                    var option = $(ev.target).find(":selected").val();
+                    if (_.has(stages, option)) {
+                        var values = stages[option];
+                        var keys = _.keys(values);
+                        for(var i=0; i<keys.length; i++) {
+                            $('#' + keys[i]).val(values[keys[i]]);
+                            $('#' + keys[i] + '_user').val(values[keys[i]]);
```
Comment (jem-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: what is the difference between `#my_id` and `#my_id_user` ? Is possible to only keep the first one ?

File: addons/planner_project/static/src/js/planner_project.js [hunk 22]
```diff
@@ -0,0 +1,125 @@
...
+                        }
+                    }
+                });
+                $( ".user_project" ).on('change', function() {
```
Comment (jem-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: Isn't that doing the same as the line 108 ?

File: addons/planner_project/static/src/js/planner_project.js (hunk 21 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: #my_id will only change current inputbox value, but we have like that, same value is stored in 'responsibility' menu.so based on select onchange both inputbox change.

File: addons/planner_project/static/src/js/planner_project.js (hunk 22 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: line 108 will only call if select box change not on inputbox change. It is required because if user change any of inputbox value, same is applied on 'responsiblity' menu inputbox.

File: addons/planner/static/src/js/planner.js [hunk 23]
```diff
@@ -182,6 +183,7 @@
             ev.preventDefault();
             var page_id = $(ev.currentTarget).attr('href').replace('#', '');
             this._switch_page(page_id);
+            $(".pages").scrollTop("0");
```
Comment (jem-odoo) [inline addons/planner/static/src/js/planner.js]: use `this.$()`

File: addons/planner_project/static/src/js/planner_project.js [hunk 24]
```diff
@@ -0,0 +1,125 @@
+(function(){
```
Comment (jem-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: use the new way to load js (from GED), with `odoo.define()`
rebase your branch.

File: addons/planner/static/src/js/planner.js (hunk 23 above)
Comment (jas-odoo) [inline addons/planner/static/src/js/planner.js]: done

File: addons/planner_project/static/src/js/planner_project.js (hunk 24 above)
Comment (jas-odoo) [inline addons/planner_project/static/src/js/planner_project.js]: Done

**Comment by rim-odoo** [comment]
Already merged at https://github.com/odoo/odoo/commit/6a26fb8087199bebe5480d46a555b96fc4e5ccfd
